cmd/mongocli/mongocli.go

```diff
@@ -96,15 +95,24 @@ func init() {
 
 	cobra.EnableCommandSorting = false
 
-	profile := rootCmd.PersistentFlags().StringP(flag.Profile, flag.ProfileShort, config.DefaultProfile, usage.Profile)
-	config.SetName(profile)
+	profile := rootCmd.PersistentFlags().StringP(flag.Profile, flag.ProfileShort, "", usage.Profile)
+	cobra.OnInitialize(func() {
+		initConfig(profile)
+	})
 }
 
 // initConfig reads in config file and ENV variables if set.
-func initConfig() {
+func initConfig(profileName *string) {
 	if err := config.Load(); err != nil {
 		log.Fatalf("Error loading config: %v", err)
 	}
+
+	availableProfiles := config.List()
+	if *profileName != "" {
+		config.SetName(profileName)
+	} else if len(availableProfiles) == 1 {
+		config.SetName(&availableProfiles[0])
+	}
```

Comment:
This sets the order to:
1. Load the configuration file
2. set the profile to a flag if set
3. if not set, see if there is only one default profile, and use that
4. stick to the default "default"